### PR TITLE
Avoid unnecessary set_pulse_register

### PR DESCRIPTION
**Pull request by rodolfocarobene**
This should help in the flipping problem, where we probably reached the maximum number of instruction per program, and overall performances should also improve.

I did not have the opportunity to test it.

With this PR qibosoq will:
- [x] Not call `set_pulse_register` when not necessary
- [x] Qick supports adding multiple_pulses **before** firing them, this should also improve performances
- [x] Adding new log file with last executed qick_program in asm (this would make debug and communications with qick easier)

Checklist before merge:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

**Comment by rodolfocarobene** [comment]
This has been deployed now on the board, I'll wait for review and some testing before merge

File: src/qibosoq/qick_programs.py
Comment (rodolfocarobene) [inline src/qibosoq/qick_programs.py:233]: hi @stavros11, in this function I wanted to compare 2 pulses (but not the start time). 
Everything worked fine, expect the comparison between shapes that, for identical rectangular pulses was giving
`pulse_a.shape == pulse_b.shape` equal to `False`.
Do you know what could have cause it?

**Comment by rodolfocarobene** [comment]
@JavierSerranoGarcia could you have a look at this? If it's ok we can merge and reboot the server today/tomorrow

**Review by JavierSerranoGarcia — APPROVED** [review]
Looks everything OK

**Pull request merged**
